packages/core-js/internals/fix-regexp-well-known-symbol-logic.js

```diff
@@ -75,7 +83,9 @@ module.exports = function (KEY, length, exec, sham) {
   if (
     !DELEGATES_TO_SYMBOL ||
     !DELEGATES_TO_EXEC ||
-    (KEY === 'replace' && !(REPLACE_SUPPORTS_NAMED_GROUPS && REPLACE_KEEPS_$0)) ||
+    (KEY === 'replace' && !(REPLACE_SUPPORTS_NAMED_GROUPS && (
+      REPLACE_KEEPS_$0 || REGEXP_REPLACE_SUBSTITUTES_UNDEFINED_CAPTURE
```

Comment:
A strange logic. Why `||`?

Reply:
My thinking was that `REPLACE_KEEPS_$0` is intended for lte IE11 but IE11 doesn't have the issue that `REGEXP_REPLACE_SUBSTITUTES_UNDEFINED_CAPTURE` intends to resolve and vice-versa, so the conditional would end up false for both of them?

Reply:
It's not something related to specific engines, it's related to specific bugs and implementations of old spec versions. At this moment we have 3 `replace`-specific criteria - so for me seems logically to use `&&` - `REPLACE_SUPPORTS_NAMED_GROUPS && REPLACE_KEEPS_$0 && !REGEXP_REPLACE_SUBSTITUTES_UNDEFINED_CAPTURE`.

Reply:
@mattclough1 Note that the `&&`s are wrapped inside `!(...)`, so the polyfill would be applied when _at least one_ of the tests is `false`